### PR TITLE
Delay the canonicalization until after processing the CLI arguments

### LINKED ISSUE #1237: OSDK should not check the options that have been overridden
### Describe the bug

OSDK will (but should not) keep checking the existence of the file, despite that option is overridden .

### To Reproduce

1. Create a `OSDK.toml` whose `initramfs` points to a non-exist file.
   ``` toml
   [boot]
   initramfs = "non/exist/directory/initramfs.cpio.gz"
   ```
2. Run OSDK with command line `initramfs` option, e.g. `cargo osdk build --initramfs="test/build/initramfs.cpio.gz"`, where `initramfs.cpio.gz` does exist.
3. OSDK will report that `non/exist/directory/initramfs.cpio.gz` does not exist, despite it's overridden by command-line `--initramfs`.
   ```
   $ cargo osdk build --initramfs="test/build/initramfs.cpio.gz"
   [Error]: Cannot canonicalize path `non/exist/directory/initramfs.cpio.gz`: No such file or directory (os error 2)
   ```

### Expected behavior

It should work, and don't need to check `non/exist/directory/initramfs.cpio.gz`.

### Screenshots

image

where `result/initrd.gz` exists while `test/build/initramfs.cpio.gz` does not.

### PR DESCRIPTION
**Pull request by StevenJiang1110**
Fixes #1237.

Originally, OSDK will check each path in OSDK.toml to ensure the path actually points to a file, but some paths that will be overrided by cli arguments should not be checked.

So this PR delays the check after processing CLI arguments.

### CODE REVIEW

**Comment by tatetian** [comment]
@junyang-zh Could you review this PR?

File: osdk/src/config/mod.rs [hunk 1]
```diff
@@ -61,7 +69,10 @@ fn apply_args_before_finalize(action_scheme: &mut ActionScheme, args: &CommonArg
             }
         }
         if let Some(initramfs) = &args.initramfs {
-            boot.initramfs = Some(initramfs.clone());
+            let initramfs = initramfs
+                .canonicalize()
+                .expect("initramfs from cli is invalid");
```
Comment (junyang-zh) [inline osdk/src/config/mod.rs:74]: Use `error_msg!` and `process::exit` to report errors to the user. Only panic in a bad program state, which could only happen if there's a bug in OSDK.

File: osdk/src/config/mod.rs [hunk 2]
```diff
@@ -73,12 +84,85 @@ fn apply_args_before_finalize(action_scheme: &mut ActionScheme, args: &CommonArg
     }
     if let Some(ref mut qemu) = action_scheme.qemu {
         if let Some(path) = &args.qemu_exe {
-            qemu.path = Some(path.clone());
+            let qemu_path = path.canonicalize().expect("qemu.path from cli is invalid");
```
Comment (junyang-zh) [inline osdk/src/config/mod.rs:87]: Here it should use `error_msg!` too. Polish the message also. A suggested one: "The QEMU path provided with argument `--qemu-exe` does not match any files."

**Review by junyang-zh — COMMENTED** [review]
A good fix! I have some minor comments.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Pull request merged**